Tweaked docs (#5)
pooltogether/ERC5164: README.md

# ERC-5164

EIP-5164 specifies how smart contracts on one chain can call contracts on another. Transport layers, such as bridges, will have their own EIP-5164 implementations. This repository includes implementations for: Ethereum to Polygon, Ethereum to Optimism, and Ethereum to Arbitrum. All three use the 'native' bridge solutions.

The EIP is currently in the Review stage: https://eips.ethereum.org/EIPS/eip-5164

Feedback and PR are welcome!

## How to use

To use ERC-5164 to send messages your contract code will need to:

- On the sending chain, send a batch of calls to the CrossChainRelayer `relayCalls` function
- Listen for calls from the corresponding CrossChainExecutor(s) on the receiving chain.

> *The listener will need to be able to unpack the original sender address (it's appended to calldata). We recommend inheriting from the included Inherit from the [`ExecutorAware.sol`](./src/abstract/ExecutorAware.sol) contract.*

Reviewer: ```suggestion
*The listener will need to be able to unpack the original sender address (it's appended to calldata). We recommend inheriting from the included [`ExecutorAware.sol`](./src/abstract/ExecutorAware.sol) contract.*
```

- [script/bridge/BridgeToOptimismGoerli.s.sol](script/bridge/BridgeToOptimismGoerli.s.sol)
- [script/bridge/BridgeToMumbai.s.sol](script/bridge/BridgeToMumbai.s.sol)

### Arbitrum Relay

> Arbitrum requires an EOA to submit a bridge transaction. The Ethereum to Arbitrum ERC5164 implementation therefore is split into two actions:

Reviewer: ```suggestion
Arbitrum requires an EOA to submit a bridge transaction. The Ethereum to Arbitrum ERC-5164 implementation therefore is split into two actions:
```

1. Calls to `relayCalls` are fingerprinted and stored along with their nonce.
2. Anyone may call `processCalls` to send a previously fingerprinted relayed call.

> The `processCalls` function same transaction parameters as the Arbitrum bridge. The [Arbitrum SDK](https://github.com/offchainlabs/arbitrum-sdk) is needed to properly estimate the gas required to execute the message on L2.

Reviewer: ```suggestion
The `processCalls` function requires the same transaction parameters as the Arbitrum bridge. The [Arbitrum SDK](https://github.com/offchainlabs/arbitrum-sdk) is needed to properly estimate the gas required to execute the message on L2.
```
